src/main/groovy/edu/oregonstate/mist/locations/frontend/resources/LocationResource.groovy

```diff
@@ -82,21 +84,109 @@ class LocationResource extends Resource {
         }
 
         String result = locationDAO.search(trimmedQ, trimmedCampus, trimmedType, pageNumber, pageSize)
-        ok(result).build()
+
+        ResultObject resultObject = new ResultObject()
+        resultObject.data = []
+
+        // parse ES into JSON Node
+        ObjectMapper mapper = new ObjectMapper() // can reuse, share globally
+        JsonNode actualObj = mapper.readTree(result)
+
+        def topLevelHits = actualObj.get("hits")
+        topLevelHits.get("hits").asList().each {
+            def source = it.get("_source")
+            ResourceObject resourceObject = new ResourceObject(
+                    id: source.get("id").asText(),
+                    type: source.get("type").asText(),
+                    attributes: source.get("attributes")
+            )
+            resultObject.data += resourceObject
+        }
+
+        setPaginationLinks(topLevelHits, q, type, campus, resultObject)
+
+        ok(resultObject).build()
+    }
+
+    /**
+     * Add pagination links to the data search results.
+     *
+     * @param topLevelHits          First "hits" node in the json document
+     * @param q
+     * @param type
+     * @param campus
+     * @param resultObject
+     */
+    private void setPaginationLinks(JsonNode topLevelHits, String q, String type, String campus,
+                                    ResultObject resultObject) {
+        def totalHits = topLevelHits.get("total").asInt()
+        // If no results were found, no need to add links
+        if (!totalHits) {
+            return
+        }
+
+        Integer pageNumber = getPageNumber()
+        Integer pageSize = getPageSize()
+        def map = ["q": q, "type": type, "campus": campus, "pageSize": pageSize, "pageNumber": pageNumber]
+        int lastPage = Math.ceil(totalHits / pageSize)
```

Comment:
verify that integer division behaves as intended. if `5 / 2 == 2`, then rounding will have no effect and a page will be missing.